api-server/src/common/models/user.json

```diff
@@ -236,6 +236,39 @@
       ],
       "default": []
     },
+    "savedChallenges": {
+      "type": [
+        {
+          "lastSavedDate": "number",
+          "id": "string",
+          "challengeType": "number",
+          "files": {
```

Comment:
Silly question: Do we want this structure to match that of the `completedChallenges.files` field? Or, do we want it to match what we want/have in the client?

Reply:
I just copied the structure we are currently using to save the javascript projects that save users code (it's right above this) - that seemed like the way to go. There's a number of fields we don't need to save to the db that are used in the client. Some of the field names are different in the client vs server - I don't particularly like that, but that's how it was being done so I stuck with it. I'm open to discussion and suggestions.